src/map/script.cpp

```diff
@@ -23696,6 +23696,27 @@ BUILDIN_FUNC(round) {
 	return SCRIPT_CMD_SUCCESS;
 }
 
+BUILDIN_FUNC(getequiptradability) {
+	int i = -1, num;
+	TBL_PC *sd;
+
+	num = script_getnum(st, 2);
+
+	if (!script_charid2sd(3, sd)) {
+		script_pushint(st, 0);
```

Comment:
never do this with line below, because in that combo you will have map-server crash when something is wrong.

Better to have something like:
```
if( !script_charid2sd(3, sd) )
    return SCRIPT_CMD_FAILURE;
```

Reply:
Older commands also do this 'combo'. (see countitem) Does it actually crash the server?

Reply:
Yes, it will. This is wrong behavior (tested). When no rid and no a "someone" to whom push int value started very strange and difficult to debug bugs and crashes of course. Do not ask me how reproduce it, because it was ~30 days+ ago. But this behavior is 100% incorrect and not need to return any value if SCRIPT_RETURN_FAILURE given. Because this constant is parsed by script engine and as I understand its a part of future PR which will present more detailed "very bad behavior" in the script what happened to map-server.

Reply:
you could `script_rid2sd(sd);`

Reply:
I agree with @anacondaqq on this one. Since we have #2063 in the works, we are going to be removing any return values on valid script failures.